pkg/artifactory/resource_artifactory_virtual_conan_repository.go

```diff
@@ -0,0 +1,43 @@
+package artifactory
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+type ConanVirtualRepositoryParams struct {
+	VirtualRepositoryBaseParams
+	VirtualRetrievalCachePeriodSecs int `hcl:"retrieval_cache_period_seconds" json:"virtualRetrievalCachePeriodSecs"`
+}
+
+func resourceArtifactoryConanVirtualRepository() *schema.Resource {
+	var conanVirtualSchema = mergeSchema(baseVirtualRepoSchema, map[string]*schema.Schema{
+		"retrieval_cache_period_seconds": {
+			Type:         schema.TypeInt,
+			Optional:     true,
+			Default:      7200,
+			Description:  "This value refers to the number of seconds to cache metadata files before checking for newer versions on aggregated repositories. A value of 0 indicates no caching.",
+			ValidateFunc: validation.IntAtLeast(0),
+		},
+	})
+
+	return mkResourceSchema(conanVirtualSchema, defaultPacker, unpackConanVirtualRepository, func() interface{} {
+		return &ConanVirtualRepositoryParams{
+			VirtualRepositoryBaseParams: VirtualRepositoryBaseParams{
+				Rclass:      "virtual",
+				PackageType: "conan",
+			},
+		}
+	})
+
+}
+
+func unpackConanVirtualRepository(s *schema.ResourceData) (interface{}, string, error) {
+	d := &ResourceData{s}
+	repo := ConanVirtualRepositoryParams{
+		VirtualRepositoryBaseParams:     unpackBaseVirtRepo(s),
+		VirtualRetrievalCachePeriodSecs: d.getInt("retrieval_cache_period_seconds", true),
```

Comment:
Also, is this field conan specific?

Reply:
This is common. Artifactory UI shows this field in different tab for few package types which confused me. But, Yes, all packages have this setting.

Reply:
Ok, so if it's common to all virtual repos, then move this code to `unpackBaseVirtualRepo` and then the only difference is that the type is `conan`

This means you should be doing this like you did for local repos. @alexhung , thoughts?

So, basically you have a generic virtual repo, that takes a param of `type` and in this base you pass it `conan` - it also means implementing anything without special fields can be done effortlessly

Reply:
Agree! If we have a group of identical virtual repos (except type), then we should create same function like the local one and build once, use many times.

Reply:
i mixed up "retrieval_cache_period_seconds" with "artifactoryRequestsCanRetrieveRemoteArtifacts".
UI, shows "retrieval_cache_period_seconds" for some package type and not for others.
So, will make this specific for Conan virtual repo.

Reply:
Created resourceArtifactoryVirtualGenericRepository function for generic like package type.
And, resourceArtifactoryVirtualRepositoryWithRetrievalCachePeriodSecs function for package types with retrieval_cache_period_seconds setting.